pbs_light_test.go

```diff
@@ -113,10 +103,70 @@ func TestCookieSyncHasCookies(t *testing.T) {
 	}
 }
 
+func TestCookieSyncNoBidders(t *testing.T) {
+	endpoint := testableEndpoint()
+
+	router := httprouter.New()
+	router.POST("/cookie_sync", endpoint)
+
+	// First test a declared empty bidders returns no syncs
+	csreq := []byte("{\"bidders\": []}")
+	csbuf := bytes.NewBuffer(csreq)
+
+	req, _ := http.NewRequest("POST", "/cookie_sync", csbuf)
+	rr := httptest.NewRecorder()
+	router.ServeHTTP(rr, req)
+	if rr.Code != http.StatusOK {
+		t.Fatalf("Wrong status: %d (%s)", rr.Code, rr.Body)
+	}
+
+	csresp := cookieSyncResponse{}
+	err := json.Unmarshal(rr.Body.Bytes(), &csresp)
+	if err != nil {
+		t.Fatalf("Unmarshal response failed: %v", err)
+	}
+
+	if csresp.Status != "no_cookie" {
+		t.Errorf("Expected status = no_cookie; got %s", csresp.Status)
+	}
+
+	if len(csresp.BidderStatus) != 0 {
+		t.Errorf("Expected 0 bidder status rows; got %d", len(csresp.BidderStatus))
+	}
+
+	// Now test a missing bidders returns all syncs
```

Comment:
You might want to split this into two tests.

Extreme "test-minded" engineers will tell you that each test should be tiny--ideally a single assertion. I've tried that, and think it's a bit overkill... but they're definitely right that short tests which check "a single thing" are easier to debug.

It basically comes down to "when a test fails, how much code do I have to read to understand to figure out what went wrong?" Methods like this double that effort.

Reply:
I had pondered splitting that myself ... so now done.